be/src/services/persistence-service.ts

```diff
@@ -15,10 +15,15 @@ export class PersistenceService {
     }
 
     private initClient(): void {
-        const MONGO_URL: string = this.errorWhenFalsy('MONGO_URL', process.env.MONGO_URL);
-        const MONGO_DATABASE: string = this.errorWhenFalsy('MONGO_DATABASE', process.env.MONGO_DATABASE);
-
-        this.client = new MongoClient(MONGO_URL, {
+        const username = process.env.MONGO_USERNAME;
+        const password = process.env.MONGO_PASSWORD;
+        const authPathParams = !!username && !!password ? `${username}:${password}@`: '';
+        const dbHost = process.env.MONGO_HOST || 'localhost';
+        const dbPort = +(process.env.MONGO_PORT || 27017);
+        const dbName = process.env.MONGO_DATABASE || 'knowledge';
+        const mongoConnect: string = process.env.MONGO_CONNECT || `mongodb://${authPathParams}${dbHost}:${dbPort}`;
```

Comment:
Please keep the name `MONGO_URL`
```suggestion
        const mongoConnect: string = process.env.MONGO_URL || `mongodb://${authPathParams}${dbHost}:${dbPort}`;
```